examples/Epidemiology/SEI3HRD_example.jl

```diff
@@ -106,18 +114,22 @@ function run_model(times::Unitful.Time, interval::Unitful.Time, timestep::Unitfu
     epilist = EpiList(traits, abun_v, abun_h, movement, transitions, param)
     rel = Gauss{eltype(epienv.habitat)}()
 
+    # multiple dispatch in action
+    rngtype = stochasticmode ? Random.MersenneTwister : MedianGenerator
+
     # Create epi system with all information
-    epi = EpiSystem(epilist, epienv, rel, scotpop)
+    epi = EpiSystem(epilist, epienv, rel, scotpop; rngtype=rngtype)
 
     # Add in initial infections randomly (samples weighted by population size)
     N_cells = size(epi.abundances.matrix, 2)
-    samp = sample(1:N_cells, weights(1.0 .* human(epi.abundances)[1, :]), 100)
+    samp = sample(ic_rng, 1:N_cells, weights(1.0 .* human(epi.abundances)[1, :]), 100)
     virus(epi.abundances)[1, samp] .= 100 # Virus pop in Environment
     human(epi.abundances)[2, samp] .= 10 # Exposed pop
 
```

Comment:
If we switch from `sample(...)` to `rand(rng, Multinomial(...))` then this will be deterministic in the same way as the epidemiology:
```suggestion
    w = weights(@view human(epi.abundances)[1, :])
    samp = rand(rngtype(), Multinomial(1000, w/sum(w)))
    virus(epi.abundances)[1, :] .+= 10 * samp # Virus pop in Environment
    human(epi.abundances)[2, :] .+= samp # Exposed pop
    human(epi.abundances)[1, :] .-= samp # Not susceptible
```